src/main/java/javaslang/control/Applicative.java

```diff
@@ -0,0 +1,17 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _ _____
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  / /  _  \   Javaslang
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/  \__/  /   Copyright 2014-now Daniel Dietrich
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/___\_/  \_//  \__/_____/    Licensed under the Apache License, Version 2.0
+ */
+package javaslang.control;
+
+public interface Applicative<T> extends Apply<T> {
```

Comment:
Well, `ap` (or `apply`) is not related to `pure` or `point`. `Pure` take a value `T` and lifts it into an `Applicative` context. Whereas `apply` takes a value already in such context, and a function also in an `Applicative` context, and applies the function to the value, returning its result still in the context. 

So I believe we need both as they perform different functions. As to whether we merge `Apply` into `Applicative` is your call. Both scalaz and cats (https://github.com/non/cats) keep the two separate. It seems cleaner to keep the conventions that both libraries use, and that would be my vote, but you tell me what you'd like to do and I'll do it. Here are links to the 2 traits in both libraries so you can take a look and decide.

Apply:
https://github.com/non/cats/blob/master/core/src/main/scala/cats/Apply.scala
https://github.com/scalaz/scalaz/blob/series/7.3.x/core/src/main/scala/scalaz/Apply.scala

Applicative:
https://github.com/non/cats/blob/master/core/src/main/scala/cats/Applicative.scala
https://github.com/scalaz/scalaz/blob/series/7.3.x/core/src/main/scala/scalaz/Applicative.scala